packages/edge-worker/src/EdgeWorker.ts

```diff
@@ -5278,4 +5303,259 @@ ${input.userComment}
 			return null;
 		}
 	}
+
+	/**
+	 * Create a Proxy that always delegates to the current raw LinearClient.
+	 *
+	 * This is the stable pointer - it never changes. On every call, it looks up
+	 * the current client from rawLinearClients Map. When token refreshes, we just
+	 * update the Map, and all existing references automatically use the new client.
+	 *
+	 * Uses promise coalescing in refreshLinearToken to handle concurrent
+	 * 401 errors - all concurrent callers wait for a single refresh.
+	 */
+	private createLinearClientProxy(repositoryId: string): LinearClient {
+		// Create a proxy that always looks up the current client
+		return new Proxy({} as LinearClient, {
```

Comment:
I hadn't seen Proxy before. I'm understanding the use case for it, but I think it probably points at an architecture pattern flaw we may have the fact that this is required? 

Maybe we should be thinking about a SelfRefreshingLinearClient class?